internal/dataapi/server/json.go

```diff
@@ -15,32 +15,25 @@
 package server
 
 import (
+	"bytes"
 	"encoding/json"
 	"fmt"
 	"io"
 
 	"github.com/FerretDB/wire/wirebson"
-	"go.mongodb.org/mongo-driver/bson"
-	"go.mongodb.org/mongo-driver/bson/bsonrw"
+	"go.mongodb.org/mongo-driver/v2/bson"
 
 	"github.com/FerretDB/FerretDB/v2/internal/util/lazyerrors"
 )
 
 // marshalJSON encodes wirebson.RawDocument into extended JSON.
 func marshalJSON(raw wirebson.RawDocument, jsonDst io.Writer) error {
-	vw, err := bsonrw.NewExtJSONValueWriter(jsonDst, false, false)
+	b, err := bson.MarshalExtJSON(bson.Raw(raw), false, false)
 	if err != nil {
 		return lazyerrors.Error(err)
 	}
 
-	encoder, err := bson.NewEncoder(vw)
-	if err != nil {
-		return lazyerrors.Error(err)
-	}
-
-	encoder.IntMinSize()
```

Comment:
@noisersup could you check  the necessity of this option please :pray:

Reply:
It's not required as far as I can see.

Overall, it seems that the usage `bson.MarshalExtJSON` seems to work here. From what I remember I had some issues with that while implementing it, but maybe some things changed between the driver versions.